### PR TITLE
Change the filter count check to a debug assertions

### PR DESCRIPTION
**Pull request by josephschorr**

### CODE REVIEW

File: pkg/spiceerrors/assert_on.go [hunk 1]
```diff
@@ -0,0 +1,12 @@
+//go:build ci
+// +build ci
+
+package spiceerrors
+
+import "fmt"
+
+func DebugAssert(condition func() bool, format string, args ...any) {
+	if !condition() {
```
Comment (ecordell) [inline pkg/spiceerrors/assert_on.go:9]: why not have the condition fn return an error?

then you can do `DebugAssert(func() {return fmt.Errorf("whatever")}`

Comment (josephschorr) [reply pkg/spiceerrors/assert_on.go:9]: I personally think this reads better: it ensures that the conditional is a boolean and we could take other actions too (like log)

Comment (ecordell) [reply pkg/spiceerrors/assert_on.go:9]: But you won't be able to return messages that depend on computation in the assert fn

Comment (josephschorr) [reply pkg/spiceerrors/assert_on.go:9]: That's fine though; its a debug assertion

File: pkg/spiceerrors/assert_off.go [hunk 2]
```diff
@@ -0,0 +1,8 @@
+//go:build !ci
+// +build !ci
+
+package spiceerrors
+
```
Comment (ecordell) [inline pkg/spiceerrors/assert_off.go:5]: Add a comment that this is always a no-op for prod builds

File: pkg/spiceerrors/assert_on.go [hunk 3]
```diff
@@ -0,0 +1,12 @@
+//go:build ci
+// +build ci
+
+package spiceerrors
+
+import "fmt"
+
```
Comment (ecordell) [inline pkg/spiceerrors/assert_on.go:7]: Add a comment that this is only enabled in CI builds (and maybe point to the other impl for prod builds)

File: pkg/spiceerrors/assert_off.go (hunk 2 above)
Comment (josephschorr) [reply pkg/spiceerrors/assert_off.go:5]: Added

File: pkg/spiceerrors/assert_on.go (hunk 3 above)
Comment (josephschorr) [reply pkg/spiceerrors/assert_on.go:7]: Added

**Review by ecordell — APPROVED** [review]

**Pull request merged**